Training5/src/task-2/src/components/user-detail/index.tsx

```diff
@@ -0,0 +1,32 @@
+import React, {useEffect, useState} from 'react';
+import {useNavigate, useParams} from "react-router-dom";
+import {User} from "../../interfaces/user.interface";
+import {useAppSelector} from "../../hook/useAppSelector";
+
+const UserDetail = () => {
+    const users = useAppSelector((state) => state.root.users.data)
+    const [userDetail, setUserDetail] = useState<User>({id: '', email: '', fullName: '', role: ''})
+    const {id} = useParams();
+    const navigate = useNavigate()
+
+    useEffect(() => {
```

Comment:
if you use useEffect for the navigation, the user still can see this page in the first render (useEffect only run after the render function), so the user can see user detail.... in short time, we should prevent it as well